src/bt_auto_dumper/_v2/__main__.py

```diff
@@ -57,49 +57,50 @@ def main(apiver: str | None = None):
 
     if not (subnet_identifier := args.subnet_identifier) or not (autovalidator_address := args.autovalidator_address):
         autovalidator_address, subnet_identifier = load_config(config_path=config_path)
-    dump_and_upload(subnet_identifier, args.subnet_realm, autovalidator_address, args.note)
+
+    wallet = bt.wallet(name="validator", hotkey="validator-hotkey", path="~/.bittensor/wallets")
+    normalized_subnet_identifier = get_normalized_codename_from_server(
+        subnet_identifier, args.subnet_realm, wallet, autovalidator_address
+    )
+    if not normalized_subnet_identifier:
+        raise ValueError("Failed to normalize the codename.")
+    dump_and_upload(normalized_subnet_identifier, args.subnet_realm, wallet, autovalidator_address, args.note)
```

Comment:
Do not ask the server to normalize it in one request and then send it back to the server in the next request.

Instead send the logs to the server specifying the subnet id or name or whatever the user will provide from the command line, or if avaiable, the value from the configuration file.